app/src/gui/components/notebook/overview_map.tsx

```diff
@@ -139,44 +140,94 @@ export const OverviewMap = (props: OverviewMapProps) => {
   mapRef.current = map;
 
-  const map_center = [30, -10];
+  const {data: map_center} = useQuery<Array<number>>({
+    queryKey: ['current_location'],
+    queryFn: async () => {
+      const position = await Geolocation.getCurrentPosition();
+      return [position.coords.longitude, position.coords.latitude];
+    },
+    initialData: [0, 0],
+  });
 
   /**
    * Create the OpenLayers map element
    */
-  const createMap = useCallback(async (element: HTMLElement): Promise<Map> => {
-    const center = transform(map_center, 'EPSG:4326', defaultMapProjection);
+  const createMap = useCallback(
+    async (element: HTMLElement): Promise<Map> => {
+      const center = transform(map_center, 'EPSG:4326', defaultMapProjection);
 
-    const tileLayer = new TileLayer({source: new OSM()});
-    const view = new View({
-      projection: defaultMapProjection,
-      center: center,
-      zoom: 12,
-    });
+      const tileLayer = new TileLayer({source: new OSM()});
+      const view = new View({
+        projection: defaultMapProjection,
+        center: center,
+        zoom: 12,
+      });
 
-    const theMap = new Map({
-      target: element,
-      layers: [tileLayer],
-      view: view,
-      controls: [new Zoom()],
-    });
+      const theMap = new Map({
+        target: element,
+        layers: [tileLayer],
+        view: view,
+        controls: [new Zoom()],
+      });
 
-    theMap.addControl(createCenterControl(theMap.getView(), center));
+      theMap.addControl(createCenterControl(theMap.getView(), center));
 
-    theMap.getView().setCenter(center);
+      theMap.getView().setCenter(center);
 
-    theMap.on('click', evt => {
-      const feature = theMap.forEachFeatureAtPixel(evt.pixel, feature => {
-        return feature.getProperties();
+      theMap.on('click', evt => {
+        const feature = theMap.forEachFeatureAtPixel(evt.pixel, feature => {
+          return feature.getProperties();
+        });
+        if (!feature) {
+          return;
+        }
+        setSelectedFeature(feature as FeatureProps);
       });
-      if (!feature) {
-        return;
-      }
-      setSelectedFeature(feature as FeatureProps);
+
+      return theMap;
+    },
+    [map_center]
+  );
+
+  /**
+   * Add a marker to the map at the current location
+   *
+   * @param map the map element
+   */
+  const addCurrentLocationMarker = (map: Map) => {
+    const source = new VectorSource();
+    const geoJson = new GeoJSON();
+
+    const stroke = new Stroke({color: 'black', width: 2});
+    const layer = new VectorLayer({
+      source: source,
+      style: new Style({
+        image: new RegularShape({
+          stroke: stroke,
+          points: 4,
+          radius: 10,
+          radius2: 0,
+          angle: 0,
+        }),
+      }),
     });
 
-    return theMap;
-  }, []);
+    const centerFeature = {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: map_center,
```

Comment:
is this safe if data is undefined - what's the behaviour?

Reply:
Have put a guard around this.